core/src/graph/key_manager.rs

```diff
@@ -337,29 +344,30 @@ mod tests {
 	}
 
 	#[test]
-	fn public_key_manager_get_key_by_id_should_return_last_key_when_duplicate_ids_exists() {
+	fn public_key_manager_get_key_by_id_should_return_first_key_when_duplicate_ids_exists() {
```

Comment:
Good question. Most probably no. But my concern was for future proofing and tolerating wrong input. 
I had 2 options either 1.panic and return an error if we had more than 1 of any key id or 2. choose the first one. 
I chose the latter just because we already had it but I'm open to change it if it would simplify anything.